### PR TITLE
InfluxDB - Add username customization

### PR DESCRIPTION
**Pull request by MilenaHC**
Adds the ability to customize username generation for dynamic users in InfluxDB.

Uses the new field `username_template` with the [go template language](https://golang.org/pkg/text/template/).

### CODE REVIEW

**Review by tomhjp — APPROVED** [review]
LGTM! 👍

**Comment by kalafut** [comment]
Nice! Will this be for 1.8? If so, please add the 1.8 Milestone to the PR.

File: plugins/database/influxdb/influxdb.go [hunk 1]
```diff
@@ -3,11 +3,11 @@ package influxdb
 import (
 	"context"
 	"fmt"
+	"github.com/hashicorp/vault/sdk/helper/template"
```
Comment (austingebauer) [inline plugins/database/influxdb/influxdb.go:6]: You may not have set this up already, but the [goimports](https://pkg.go.dev/golang.org/x/tools/cmd/goimports) tool is useful to organize imports in a consistent way. I suggest to try running that tool on this file! I would expect to see this import grouped with the other group (`"github.com/*"`) below.

This is just a stylistic suggestion. We usually try to organize imports consistently in Vault.

**Review by austingebauer — APPROVED** [review]
Looks good to me! Great job 👍

**Review by calvn — APPROVED** [review]
Same thoughts on imports as Austin mentioned, but otherwise looks great!

**Pull request merged**